src/App.tsx

```diff
@@ -1,27 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[] | []>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleGoods = async (getGoods: () => Promise<Good[]>) => {
+    const allGoodsFromServer: Good[] = await getGoods();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    try {
+      setGoods(allGoodsFromServer);
+    } catch (error) {
```

Comment:
this won't throw any error, you should «try» to fetch, saving goods must be fine
also, you throw error, and it'd be great to handle it (e.g. display it in your app)

Reply:
Done, in my opinion. I show the error on the screen.